kni-test.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+var fs = require('fs');
```

Comment:
Oh, wow. I didn’t use promises for this. How very quaint. Now `fs.promises` has promisified all the methods and we can await our way to a cleaner program.

Reply:
Yes, but that's part of modernization; I'm at least trying to keep this bug fixing sidequest more easily back portable to v4/v3 world; then we can uplift everything to promises and/or async/await.

I suppose it matters less for the test code, but I'd prefer to be able to cherry pick not just the fixes, but also their test code fwiw.

Also note the rimraf, since we can't yet take node v14 for granted, ala `fs.rm`